server/src/test/java/org/opensearch/action/support/clustermanager/TransportClusterManagerTermCheckTests.java

```diff
@@ -0,0 +1,320 @@
+        }
+
+        @Override
+        protected void clusterManagerOperation(Request request, ClusterState state, ActionListener<Response> listener) throws Exception {
+            listener.onResponse(new Response()); // default implementation, overridden in specific tests
+        }
+
+        @Override
+        protected ClusterBlockException checkBlock(Request request, ClusterState state) {
+            return null; // default implementation, overridden in specific tests
+        }
+    }
+
+    public void testTermCheckMatchWithClusterManager() throws ExecutionException, InterruptedException {
+        setUpCluster(Version.CURRENT);
+
+        TransportClusterManagerTermCheckTests.Request request = new TransportClusterManagerTermCheckTests.Request();
+        PlainActionFuture<TransportClusterManagerTermCheckTests.Response> listener = new PlainActionFuture<>();
+        new TransportClusterManagerTermCheckTests.Action("internal:testAction", transportService, clusterService, threadPool).execute(
+            request,
+            listener
+        );
+
+        assertThat(transport.capturedRequests().length, equalTo(1));
+        CapturingTransport.CapturedRequest capturedRequest = transport.capturedRequests()[0];
+        assertTrue(capturedRequest.node.isClusterManagerNode());
+        assertThat(capturedRequest.action, equalTo("cluster:monitor/term"));
+        GetTermVersionResponse response = new GetTermVersionResponse(
+            new ClusterStateTermVersion(
+                clusterService.state().getClusterName(),
+                clusterService.state().metadata().clusterUUID(),
+                clusterService.state().term(),
+                clusterService.state().version()
+            )
+        );
+        transport.handleResponse(capturedRequest.requestId, response);
+        assertTrue(listener.isDone());
+    }
+
+    public void testTermCheckNoMatchWithClusterManager() throws ExecutionException, InterruptedException {
+        setUpCluster(Version.CURRENT);
+        TransportClusterManagerTermCheckTests.Request request = new TransportClusterManagerTermCheckTests.Request();
+
+        PlainActionFuture<TransportClusterManagerTermCheckTests.Response> listener = new PlainActionFuture<>();
+        new TransportClusterManagerTermCheckTests.Action("internal:testAction", transportService, clusterService, threadPool).execute(
+            request,
+            listener
+        );
+
+        assertThat(transport.capturedRequests().length, equalTo(1));
+        CapturingTransport.CapturedRequest termCheckRequest = transport.capturedRequests()[0];
+        assertTrue(termCheckRequest.node.isClusterManagerNode());
+        assertThat(termCheckRequest.action, equalTo("cluster:monitor/term"));
+        GetTermVersionResponse termVersionResponse = new GetTermVersionResponse(
+            new ClusterStateTermVersion(
+                clusterService.state().getClusterName(),
+                clusterService.state().stateUUID(),
```

Comment:
`clusterUUID` ?
Should we also have some assertion on the response, may be in integ test ?

Reply:
changed to clusterUUID.

Tests asserts based on ClusterState response and call count invocations since `GetTermVersionResponse` is internal and not exposed through client for assertion. 

This is a mismatch test to return out-of-sync response and hence call count did not change for assert to fail :) .